src/main/java/entities/Tag.java

```diff
@@ -18,24 +24,26 @@ public class Tag {
     Color color;
 
     /**
-     * The User (owner) of the tag
+     * The unique identifier of the tag
      */
-//    final User user;
+    String uuid;
 
     /**
      * Instantiates a new Tag with a name and colour
-     * @param name The name of the Tag
+     *
+     * @param name  The name of the Tag
      * @param color The colour of the Tag
-     * // @param user The owner of the Tag
+     *              // @param user The owner of the Tag
      */
-    public Tag(String name, Color color /* User user */) {
+    public Tag(String name, Color color, User user) {
```

Comment:
I haven't reviewed the code from the "autograding" perspective last time, but on the autograde test this threw an Error because the CreateTag class called this method without the User parameter. One solution would be to implement an overloaded method with the User set to null and assign a User later, or otherwise to pass a User into CreateTag (either as an instance variable or a param) and do it without overloading.